src/parser/parser.cpp

```diff
@@ -0,0 +1,170 @@
+/******************************************************************************
+ * Copyright (C) 2022 by Gschwind, Weber, Schoch, Niederberger.               *
+ *                                                                            *
+ * This file is part of linescaleGUI.                                         *
+ *                                                                            *
+ * LinescaleGUI is free software: you can redistribute it and/or modify       *
+ * it under the terms of the GNU General Public License as published by       *
+ * the Free Software Foundation, either version 3 of the License, or          *
+ * (at your option) any later version.                                        *
+ *                                                                            *
+ * LinescaleGUI is distributed in the hope that it will be useful,            *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the               *
+ * GNU General Public License for more details.                               *
+ *                                                                            *
+ * You should have received a copy of the GNU General Public License          *
+ * along with LinescaleGUI.  If not, see <http://www.gnu.org/licenses/>.      *
+ ******************************************************************************/
+/**
+ * @file parser.cpp
+ * @authors Gschwind, Weber, Schoch, Niederberger
+ * @copyright (c) 2022 LineGrip Corp Ltd.
+ *
+ * @brief Parses the line scale communication protocol
+ *
+ */
+
+#include "parser.h"
+
+bool Parser::parsePackage(QByteArray& package, DataStruct& data) {
+    bool dataOk = checkPackage(package);
+
+    if (dataOk) {
+        dataOk = parseWorkingMode(package, data); 
+    }
+    if (dataOk) {
+        parseMeasuredValue(package, data);  
+        dataOk = parseMeasureMode(package, data);
+    }
+    if (dataOk) {
+        parseReferenceZero(package, data);
+        parseBatteryPercent(package, data);
+        dataOk = parseUnitValue(package, data);
+    }
+    if (dataOk) {
+        dataOk = parseFrequency(package, data);
+    }
+    return dataOk;
+}
+
+bool Parser::checkPackage(QByteArray& package) {
+    QByteArray value;
+    int checkVal = 0;
```

Comment:
General coding style: declare the variables at the point you're going to use them (i.e. before the `for` loop).
We're not coding C here (isn't even required in C since a long time).